packages/amd-k8s-device-plugin/amd-k8s-device-plugin.service

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Start AMD kubernetes device plugin
+After=kubelet.service
+Wants=kubelet.service
+ConditionPathExists=/usr/bin/ghostdog
+
+[Service]
+# Verify AMD GPU is detected
+ExecStartPre=/usr/bin/ghostdog amd-gpu-present
+# Verify amdgpu driver is loaded
+ExecStartPre=/usr/bin/ghostdog match-driver amd rocm
+# Ensure that the kubelet device plugin socket exists before we start
+# A brief sleep is needed to avoid the `test` failing its first check
+ExecStartPre=/usr/bin/sleep 0.1
```

Comment:
This is a sign your dependencies are incorrectly wired up. Do you actually need kubelet.service or the socket, if you need the socket, let's make that the dependency of this unit.